lib/mirage/mirage.ml

```diff
@@ -295,6 +295,22 @@ let git_http ?authenticator ?headers ?(pclock = default_posix_clock) tcpv4v6 ctx
     =
   Mirage_impl_git.git_http ?authenticator headers $ pclock $ tcpv4v6 $ ctx
 
+let delay = job
+
+let delay_startup =
+  let delay_key = Key.delay in
+  let keys = [ Key.v delay_key ] in
+  let packages = [ package ~max:"1.0.0" "duration" ] in
+  let connect i _ _ =
+    let modname = match Mirage_impl_misc.get_target i with
+      | `Unix | `MacOSX -> "Unix_os.Time"
+      | `Xen | `Qubes -> "Xen_os.Time"
+      | `Virtio | `Hvt | `Spt | `Muen | `Genode -> "Solo5_os.Time"
```

Comment:
this duplicates some code, but using a functor imposes that we'd need a module to apply the Time onto, and I don't think it is worth it. This way, it is a rather slick implementation (of course, we now have the information about the Time module name duplicated in mirage_impl_time and here -- happy to hear suggestions that don't require us to have a functor around).